eng/common/scripts/Test-Spelling.ps1

```diff
@@ -0,0 +1,87 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $TargetRef = 'master', 
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $CspellConfigPath = "./.vscode/cspell.json"
+)
+
+. $PSScriptRoot/logging.ps1
+
+if ((Get-Command git | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate git. Install git https://git-scm.com/downloads"
+    exit 1
+}
+
+if ((Get-Command npx | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate npx. Install NodeJS (includes npm and npx) https://nodejs.org/en/download/"
+    exit 1
+}
+
+$initialDirectory = Get-Location
+$exitCode = 0
+try { 
+    Set-Location "$PSScriptRoot/../../.."
+
+    # Lists names of files that were in some way changed between the 
+    # current ref and $TargetRef. Excludes files that were deleted to
+    # prevent errors in Resolve-Path
+    Write-Host "git diff --diff-filter=d --name-only $TargetRef"
+    $changedFiles = git diff --diff-filter=d --name-only $TargetRef `
+        | Resolve-Path
+    
+    $changedFilesCount = ($changedFiles | Measure-Object).Count
+    Write-Host "Git Detected $changedFilesCount changed file(s). Files checked by cspell may exclude files according to cspell.json"
+
+    if ($changedFilesCount -eq 0) {
+        Write-Host "No changes detected"
+        # The finally block still runs after calling exit here
+        exit $exitCode
+    }
+
+    $changedFilesString = $changedFiles | Join-String -Separator ' '
```

Comment:
```suggestion
    $changedFilesString = $changedFiles -join ' '
```
While amounts to the same there is a join operator (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_join?view=powershell-7.1) which is a little more efficient since it doesn't require the use of the object pipeline.